llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -1355,7 +1355,7 @@ SmallPtrSet<Type *, 4> collectGenXVolatileTypes(Module &M) {
       continue;
     assert(GTy->getNumContainedTypes() == 1);
     auto VTy = GTy->getContainedType(0);
-    if (GTy = dyn_cast<StructType>(VTy)) {
+    if ((GTy = dyn_cast<StructType>(VTy))) {
```

Comment:
@AGindinson, could you address [this comment](https://github.com/intel/llvm/commit/01351f11854ae4ee86e940cd0249b5d5a7865d08#r57524435) while you are here, please?

Reply:
Sure, done!